### PR TITLE
yaac: init at 1.0-beta182

### PR DESCRIPTION
**Pull request by dotemup**
###### Description of changes

New application YAAC, Yet Another APRS Client. This application is a Radio/Network application for sending and receiving packets through the AX.25 protocol.

Tested in configurations with raw AX.25 I/O, and through assistance with Direwolf, and hamlib rigctld via FLRIG.

Added myself as new maintainer and added new application into top-level call, correctly dist into radio sub-directory.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix
Comment (natsukium) [inline maintainers/maintainer-list.nix:1]: Would you please split this change into separate commits?
The commit message should be `maintainers: add dotemup`.

File: pkgs/applications/radio/yaac/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ makeWrapper, lib, stdenv, fetchurl, unzip, jre8 }:
```
Comment (natsukium) [inline pkgs/applications/radio/yaac/default.nix:1]: One item per line is preferable for future diff.

File: pkgs/applications/radio/yaac/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+stdenv.mkDerivation rec {
+  version = "1.0-beta181";
+  pname = "YAAC";
```
Comment (natsukium) [inline pkgs/applications/radio/yaac/default.nix:5]:
```suggestion
  pname = "yaac";
```
> The pname and the version attribute must not contain uppercase letters

https://nixos.org/manual/nixpkgs/stable/#sec-package-naming

File: pkgs/applications/radio/yaac/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  src = fetchurl {
```
Comment (natsukium) [inline pkgs/applications/radio/yaac/default.nix:7]: How about using `fetchzip` here?

File: pkgs/applications/radio/yaac/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    url = "https://www.ka2ddo.org/ka2ddo/YAAC.zip";
+    sha256 = "564071bb51fb4a91850ff53ff7d0c9477c1715d7b11eed9e97c89ead9df7332b";
```
Comment (natsukium) [inline pkgs/applications/radio/yaac/default.nix:9]: Please use SRI hash here.
> Nixpkgs contributors are currently recommended to use hash.

https://nixos.org/manual/nixpkgs/stable/#fetchurl

File: pkgs/applications/radio/yaac/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  };
+
+  nativeBuildInputs = [ unzip makeWrapper ];
+
+  unpackPhase = "unzip $src -d $out";
+
+  # TODO: Make Desktop Icon
+
+  installPhase = ''
```
Comment (natsukium) [inline pkgs/applications/radio/yaac/default.nix:18]: Remember to call `runHook` when overwriting a phase.
https://nixos.org/manual/nixpkgs/stable/#sec-stdenv-phases

File: pkgs/applications/radio/yaac/default.nix [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    makeWrapper ${jre8}/bin/java $out/bin/yaac \
+      --add-flags "-jar $out/YAAC.jar" \
+      --set _JAVA_OPTIONS '-Dawt.useSystemAAFontSettings=on' \
+      --set _JAVA_AWT_WM_NONREPARENTING 1
+  '';
+
+  meta = with lib; {
+    description = "Yet Another APRS Client";
+    homepage = "https://www.ka2ddo.org/ka2ddo/YAAC.html";
+    license = licenses.lgpl3Only;
```
Comment (natsukium) [inline pkgs/applications/radio/yaac/default.nix:29]:
```suggestion
    license = licenses.lgpl3Plus;
```
The source code looks like lgpl3Plus.
https://sourceforge.net/p/yetanotheraprsc/code/HEAD/tree/trunk/yaac/YAAC/src/main/java/org/ka2ddo/yaac/YAAC.java#l9

File: pkgs/applications/radio/yaac/default.nix [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    platforms = platforms.linux;
```
Comment (natsukium) [inline pkgs/applications/radio/yaac/default.nix:30]:
```suggestion
    platforms = platforms.unix;
```
It seems to work in any environment where java runs, at least on unix.

File: maintainers/maintainer-list.nix
Comment (dotemup) [reply maintainers/maintainer-list.nix:1]: Done. https://github.com/NixOS/nixpkgs/pull/242772

Comment (dotemup) [reply maintainers/maintainer-list.nix:1]: Should a similar separate commit be pushed for the changes to pkgs/top-level/all-packages.nix?

File: pkgs/applications/radio/yaac/default.nix (hunk 1 above)
Comment (dotemup) [reply pkgs/applications/radio/yaac/default.nix]: Updated in latest.

File: maintainers/maintainer-list.nix
Comment (dotemup) [reply maintainers/maintainer-list.nix:1]: Updated in latest.

File: pkgs/applications/radio/yaac/default.nix (hunk 2 above)
Comment (dotemup) [reply pkgs/applications/radio/yaac/default.nix]: Updated in latest.

File: pkgs/applications/radio/yaac/default.nix (hunk 4 above)
Comment (dotemup) [reply pkgs/applications/radio/yaac/default.nix]: Updated in latest.

File: pkgs/applications/radio/yaac/default.nix (hunk 5 above)
Comment (dotemup) [reply pkgs/applications/radio/yaac/default.nix:22]: Updated in latest.

File: pkgs/applications/radio/yaac/default.nix (hunk 7 above)
Comment (dotemup) [reply pkgs/applications/radio/yaac/default.nix]: Updated in latest.

File: pkgs/applications/radio/yaac/default.nix (hunk 6 above)
Comment (dotemup) [reply pkgs/applications/radio/yaac/default.nix]: Updated in latest.

File: pkgs/applications/radio/yaac/default.nix (hunk 3 above)
Comment (dotemup) [reply pkgs/applications/radio/yaac/default.nix]: Updated, didn't use initially as wasn't aware. Was tricky with the stripRoot at first. but got it working.

File: maintainers/maintainer-list.nix [hunk 8]
```diff
@@ -4297,6 +4297,11 @@
     githubId = 10998835;
     name = "Doron Behar";
   };
+  dotemup = {
+    name = "Dote";
+    github = "dotemup";
+    githubId = 11077277;
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:4303]: It would be nice to have another mean of contact

File: pkgs/applications/radio/yaac/default.nix [hunk 9]
```diff
@@ -0,0 +1,46 @@
+{ fetchzip
+, jre8
+, lib
+, makeWrapper
+, stdenv
+}:
+
+stdenv.mkDerivation rec {
+  version = "1.0-beta181";
+  pname = "yaac";
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/yaac/default.nix:10]:
```suggestion
  pname = "yaac";
  version = "1.0-beta181";
```

File: pkgs/applications/radio/yaac/default.nix [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+  src = fetchzip {
+    url = "https://www.ka2ddo.org/ka2ddo/YAAC.zip";
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/yaac/default.nix:13]: DO we have a versioned download available? otherwise please upload it to web.archive.org

File: pkgs/applications/radio/yaac/default.nix [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+    hash = "sha256-15oH+I/ACbcHQ6bMohKoO1+GKDu8ThkjKTKN0JIHty8=";
+    stripRoot = false;
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  # TODO: Make Desktop Icon
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/yaac/default.nix:20]: Why not do this now?

File: pkgs/applications/radio/yaac/default.nix [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/yaac/default.nix:25]:
```suggestion
```

File: pkgs/applications/radio/yaac/default.nix (hunk 10 above)
Comment (natsukium) [reply pkgs/applications/radio/yaac/default.nix:13]: I had overlooked that this is not versioned. Thanks.

There is the source code managed by svn in sourceforge.
https://sourceforge.net/p/yetanotheraprsc/code/HEAD/tree/

How about using this?

Comment (dotemup) [reply pkgs/applications/radio/yaac/default.nix:13]: Doing an ant build now with fetchsvn, it's a little funky with the way it's unpacked now though. Reinstating unzip in install phase to unpack the zip the ant build creates. Appears to be working, just need to test the application to ensure it works as expected for usage.

File: pkgs/applications/radio/yaac/default.nix (hunk 11 above)
Comment (dotemup) [reply pkgs/applications/radio/yaac/default.nix:20]: Not sure of the best way to achieve it. 

There's a 32 and 64 ico file in the images dir in the output, I just haven't had much luck in creating it with the makeDesktopItem/copyDesktopItems hooks in my limited attempts at it. Also not sure of disposing of them when the package is removed, I've had some nix-env app icons installed that persist after uninstall.

The binary created is mostly what's used but a desktop icon is just a nice to have.

File: maintainers/maintainer-list.nix (hunk 8 above)
Comment (dotemup) [reply maintainers/maintainer-list.nix:4303]: Updated in latest.

File: pkgs/applications/radio/yaac/default.nix (hunk 9 above)
Comment (dotemup) [reply pkgs/applications/radio/yaac/default.nix]: Updated in latest.

File: pkgs/applications/radio/yaac/default.nix (hunk 12 above)
Comment (dotemup) [reply pkgs/applications/radio/yaac/default.nix]: Updated in latest.

File: pkgs/applications/radio/yaac/default.nix (hunk 10 above)
Comment (dotemup) [reply pkgs/applications/radio/yaac/default.nix]: Rebuilt locally on my radio station, works as expected built from source. However the build takes a long time, and I'm pretty sure there's something to optimize here now. Will take another look tomorrow.

**Comment by dotemup** [comment]
Updated with combined commit for adding package to top-level.

File: pkgs/applications/radio/yaac/default.nix [hunk 13]
```diff
@@ -0,0 +1,48 @@
+{ ant, fetchsvn, jdk, jre8, lib, makeWrapper, stdenv, unzip }:
+
+stdenv.mkDerivation rec {
+
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/yaac/default.nix:4]:
```suggestion
```

File: pkgs/applications/radio/yaac/default.nix [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+  pname = "yaac";
+  version = "1.0-beta182";
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/yaac/default.nix:6]:
```suggestion
  version = "229";
```
Would that be more correct? I am not sure

File: pkgs/applications/radio/yaac/default.nix [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+
+  src = fetchsvn {
+    url = "https://svn.code.sf.net/p/yetanotheraprsc/code/trunk/yaac";
+    rev = "r229";
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/yaac/default.nix:10]:
```suggestion
    rev = "r${version}";
```

File: pkgs/applications/radio/yaac/default.nix (hunk 11 above)
Comment (SuperSandro2000) [reply pkgs/applications/radio/yaac/default.nix:16]: > Also not sure of disposing of them when the package is removed, I've had some nix-env app icons installed that persist after uninstall.

that's on nix-env maybe leaving symlinks behind, nothing nixpkgs can really fix.



> I just haven't had much luck in creating it with the makeDesktopItem/copyDesktopItems hooks in my limited attempts at it.

I can't really provide you with help if I don't have more details.

File: pkgs/applications/radio/yaac/default.nix (hunk 14 above)
Comment (dotemup) [reply pkgs/applications/radio/yaac/default.nix:6]: I don't believe so, the versioning of the application is different from the sourceforge revision.

image

**Comment by dotemup** [comment]
Closing request while working with application owner to resolve issues with build
